src/main/scala/com/redhat/et/silex/statistics/cramersv.scala

```diff
@@ -0,0 +1,120 @@
+/*
+ * cramersv.scala
+ * author:  RJ Nowling <rnowling@redhat.com>
+ *
+ * Copyright (c) 2016 Red Hat, Inc.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.redhat.et.silex.statistics
+
+import scala.util.Random
+
+/**
+ * "[[https://en.wikipedia.org/wiki/Cram%C3%A9r%27s_V Cramers' V]] is a measure of
+ * association between two nominal variables, giving a value between 0 and +1
+ * (inclusive)." 
+ */
+object CramersV {
+  /**
+   * Calculate Cramer's V for two sets (`values1` and `values2`) of co-sampled values.
+   */
+  def apply[T, U](values1 : Seq[T], values2 : Seq[U]) : Double = {
+    val set1 = values1.toSet
+    val set2 = values2.toSet
+
+    if (set1.size == 1 && set2.size == 1) {
+      1.0
+    } else if (values1.size == 0 || set1.size == 1 || set2.size == 1) {
+      0.0
+    } else {
+      val pairCounts = values1.zip(values2)
+        .foldLeft(Map.empty[(T, U), Double]) {
+        case (counts, value) =>
+          counts + (value -> (counts.getOrElse(value, 0.0) + 1.0))
+      }
+
+      val counts1 = values1
+        .foldLeft(Map.empty[T, Double]) {
+        case (counts, value) =>
+          counts + (value -> (counts.getOrElse(value, 0.0) + 1.0))
+      }
+
+      val counts2 = values2
+        .foldLeft(Map.empty[U, Double]) {
+        case (counts, value) =>
+          counts + (value -> (counts.getOrElse(value, 0.0) + 1.0))
+      }
+
+      val nObs = values1.size.toDouble
+
+      val combinations = set1.flatMap {
+        v1 =>
+          set2.map {
+            v2 =>
+              (v1, v2)
+          }
+      }
+      .toSeq
+
+      val chi2 = combinations.map {
+        case (value1, value2) =>
+          val nij = pairCounts.getOrElse((value1, value2), 0.0)
+          val ni = counts1.getOrElse(value1, 0.0)
+          val nj = counts2.getOrElse(value2, 0.0)
+
+          val b = ni * nj / nObs
+          val c = (nij - b) * (nij - b) / b
+
+          c / nObs
+      }
+      .sum
```

Comment:
Because you are just summing here, consider using a fold/foldLeft here.   Map actually instantiates a new sequence.  

Another technique is to convert to iterator asap, which also avoids instantiating entire collections.  I actually wrote a little function `cross` to generate crossproducts as an iterator, which can be helpful because it avoids instantiating the entire crossproduct of pairs at any one time:

https://github.com/erikerlandson/unit4s/blob/unit9a/codegen/src/main/scala/com/manyangled/codegen/codegen.scala#L9

So you could do `cross(set1, set2).map(...).sum`   or `cross(set1, set2).fold(0.0)(...)`